File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        newstr = ""
        for i in range(len(strs[0])):
            for c in strs:
                if i == len(c) or c[i]!= strs[0][i]:
                    return newstr
            newstr += strs[0][i]
                    
        return newstr
            

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        String str1 = strs[0];
        String str2 = strs[strs.length-1];
        int indx = 0;
        while(indx < str1.length() && indx < str2.length()){
            if(str1.charAt(indx) == str2.charAt(indx)){
                indx++;
            } else {
                break;
            }
        }
        return str1.substring(0, indx);
    }
}

